earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp

```diff
@@ -23,14 +23,343 @@ namespace {
+    const std::string expectedJson = R"js_({"name":"fred.flintstone"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// this is legal in json
+TEST_F(JsonUnitTest, dashInKey) {
+    const std::string js = R"js_({"my-name-is":"fred"})js_";
+    const std::string expectedJson = R"js_({"my-name-is":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// this is legal in json
+TEST_F(JsonUnitTest, dashInValue) {
+    const std::string js = R"js_({name:"fred-flintstone"})js_";
+    const std::string expectedJson = R"js_({"name":"fred-flintstone"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// this is legal in json
+TEST_F(JsonUnitTest, spaceStartValue) {
+    const std::string js = R"js_({" name":"fred"})js_";
+    const std::string expectedJson = R"js_({" name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// This is legal in json
+TEST_F(JsonUnitTest, underscoreStartValue) {
+    const std::string js = R"js_({"_name":"fred"})js_";
+    const std::string expectedJson = R"js_({"_name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// this is legal in json
+TEST_F(JsonUnitTest, dotStartValue) {
+    const std::string js = R"js_({".name":"fred"})js_";
+    const std::string expectedJson = R"js_({".name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// this is legal in json
+TEST_F(JsonUnitTest, dashStartValue) {
+    const std::string js = R"js_({"-name":"fred"})js_";
+    const std::string expectedJson = R"js_({"-name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// While illegal in json with . reference it is ok otherwise
+TEST_F(JsonUnitTest, numericStartValue) {
+    const std::string js = R"js_({"123name":"fred"})js_";
+    const std::string expectedJson = R"js_({"123name":"fred"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// Simple multiple pair case
+TEST_F(JsonUnitTest, multiplePairs) {
+    const std::string js = R"js_({name:"fred",car:"christine"})js_";
+    const std::string expectedJson = 
+        R"js_({"name":"fred","car":"christine"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// Should it eat white space? I would think the conversion would collapse it.
+// This may not be a bug.  Yes the test is ugly on purpose to contrast output
+//TEST_F(JsonUnitTest, spansMultipleLines) {
+//    const std::string js = R"js_({name:"fred", 
+//         car:"christine"
+//        ,color:"red"})js_";
+//    const std::string expectedJson = 
+//        R"js_({"name":"fred","car":"christine","color":"red"})js_";

```

Comment:
Yea I wasn't clear on that.  I think currently it preserves white space.  Maybe we can add an option to create compact JSON.  @tst-jhaight can confirm behavior.